components/formats-gpl/src/loci/formats/in/ImprovisionTiffReader.java

```diff
@@ -162,31 +165,58 @@ protected void initStandardMetadata() throws FormatException, IOException {
 
     put("Improvision", "yes");
 
-    // parse key/value pairs in the comment
-    String comment = ifds.get(0).getComment();
+    // parse key/value pairs in the comments
+    String[] comments = new String[ifds.size()];
     String tz = null, tc = null, tt = null;
-    if (comment != null) {
-      String[] lines = comment.split("\n");
-      for (String line : lines) {
-        int equals = line.indexOf('=');
-        if (equals < 0) continue;
-        String key = line.substring(0, equals);
-        String value = line.substring(equals + 1);
-        addGlobalMeta(key, value);
-        if (key.equals("TotalZPlanes")) tz = value;
-        else if (key.equals("TotalChannels")) tc = value;
-        else if (key.equals("TotalTimepoints")) tt = value;
-        else if (key.equals("XCalibrationMicrons")) {
-          pixelSizeX = DataTools.parseDouble(value);
-        }
-        else if (key.equals("YCalibrationMicrons")) {
-          pixelSizeY = DataTools.parseDouble(value);
-        }
-        else if (key.equals("ZCalibrationMicrons")) {
-          pixelSizeZ = DataTools.parseDouble(value);
+    for (int plane=0; plane<ifds.size(); plane++) {
+      String comment = ifds.get(plane).getComment();
+      comments[plane] = comment;
+      if (comment != null) {
+        String[] lines = comment.split("\n");
+        for (String line : lines) {
+          int equals = line.indexOf('=');
+          if (equals < 0) continue;
+          String key = line.substring(0, equals);
+          String value = line.substring(equals + 1);
+          addGlobalMeta(key, value);
+          if (key.equals("TotalZPlanes")) tz = value;
+          else if (key.equals("TotalChannels")) tc = value;
+          else if (key.equals("TotalTimepoints")) tt = value;
+          else if (key.equals("XCalibrationMicrons")) {
+            pixelSizeX = DataTools.parseDouble(value);
+          }
+          else if (key.equals("YCalibrationMicrons")) {
+            pixelSizeY = DataTools.parseDouble(value);
+          }
+          else if (key.equals("ZCalibrationMicrons")) {
+            pixelSizeZ = DataTools.parseDouble(value);
+          }
+          else if (key.equals("WhiteColour")) {
+            String[] rgb = value.split(",");
+            if (rgb.length < 3) {
+              channelColors.add(null);
+              continue;
+            }
+            int red = 255;
+            try {
+              red = Integer.parseInt(rgb[0]);
```

Comment:
As long as the default value remains optional, I think that would be a useful addition.